### PR TITLE
Use log instead of println for logging

### LINKED ISSUE #117: Proper logging in pruntime & phost
Refactor `phost` and `pruntime` subprojects. Use `log` crate instead of wild `println!` everywhere. On the binary side we can use [env_logger](https://docs.rs/env_logger/0.8.3/env_logger/) as the output.

Deliverables:

1. Convert all the `println!` to log, based on your judgement of the log level, unless special cases (we can communicate)
2. Test the full stack (phala-node, phost, pruntime) and make sure they can build, and sync to at least 1k blocks.

Prepare the dev env: https://wiki.phala.network/en-us/docs/developer/run-a-local-development-network/
  - Mostly up-to-date, but pruntime & phost have been moved to standalone

### PR DESCRIPTION
**Pull request by KiChjang**
Fixes #117.

### CODE REVIEW

**Comment by h4x3rotab** [comment]
image

Nice job! The `phost` log works like a charm. However it looks like `pruntim` logs inside the enclave is not correctly populated to the console. [This doc](https://github.com/apache/incubator-teaclave-sgx-sdk/blob/07adab206e7507c2ea4c41c6ba2a1037015e852b/samplecode/logger/Readme.md) looks like a solution.

**Comment by KiChjang** [comment]
@h4x3rotab Slight problem -- when I try to add `env_logger` as a git dependency per the instructions from the document, the build fails for pRuntime:

```
$ SGX_MODE=SW make

// Rust compilation stuff

    Finished release [optimized] target(s) in 32.55s
cp ./target/release/libenclaveapp.a ../lib/libenclave.a
make[1]: Leaving directory '/d/Workspace/phala-blockchain/standalone/pruntime/enclave'
/usr/bin/ld: ./lib/libenclave.a(sgx_libc-9e53db721c849504.sgx_libc.dvoihyds-cgu.8.rcgu.o): in function `sgx_libc::linux::x86_64::ocall::getenv':
sgx_libc.dvoihyds-cgu.8:(.text._ZN8sgx_libc5linux6x86_645ocall6getenv17hce591463a4d235e3E+0x11): undefined reference to `u_getenv_ocall'
collect2: error: ld returned 1 exit status
make: *** [Makefile:175: enclave/enclave.so] Error 1
```

I googled around for `undefined reference to u_getenv_ocall` and found an issue that may be related: https://github.com/scs/substraTEE-worker/issues/59

I'm going to push my changes as a new commit to this branch. How should I proceed?

**Comment by h4x3rotab** [comment]
We need to add this line:

https://github.com/apache/incubator-teaclave-sgx-sdk/blob/07adab206e7507c2ea4c41c6ba2a1037015e852b/samplecode/logger/enclave/Enclave.edl#L37

**Comment by h4x3rotab** [comment]
Just noticed `env::vars()` is empty inside the enclave. Not sure why

**Comment by KiChjang** [comment]
> 
> 
> Just noticed `env::vars()` is empty inside the enclave. Not sure why. However it outputs if we skip the environment variable and init env_logger directly:
> 
> ```
>     env_logger::builder()
>         .filter(None, log::LevelFilter::Trace)
>         .init();
> ```

Hmm... should I go ahead and make this change? It's quite suboptimal that we can't change the log level via environment variables though... perhaps we can pass in an argument somehow?

**Comment by h4x3rotab** [comment]
> > Just noticed `env::vars()` is empty inside the enclave. Not sure why. However it outputs if we skip the environment variable and init env_logger directly:
> > ```
> >     env_logger::builder()
> >         .filter(None, log::LevelFilter::Trace)
> >         .init();
> > ```
> 
> Hmm... should I go ahead and make this change? It's quite suboptimal that we can't change the log level via environment variables though... perhaps we can pass in an argument somehow?

Let's make it in this way. I will open another bounty for a total fix.

**Comment by h4x3rotab** [comment]
Oh, don't forget to initialize `env_logger` at pruntime/app/src/main.rs

**Comment by h4x3rotab** [comment]
LOL. I tried to add `env_logger` in `pruntime/app`, and guess what? The environment variables then can pass to the enclave correctly!

Please add `env_logger` and init it in `pruntime/app/src/main.rs`, and then check if the outputs from the enclave can show in the terminal. (I've tested it and it worked locally)

File: standalone/pruntime/enclave/src/lib.rs [hunk 1]
```diff
@@ -979,6 +974,8 @@ fn init_runtime(input: InitRuntimeReq) -> Result<Value, Value> {
         return Err(json!({"message": "Already initialized"}));
     }
 
+    env_logger::init();
```
Comment (KiChjang) [inline standalone/pruntime/enclave/src/lib.rs:977]: @h4x3rotab Should I remove this init then?

Comment (h4x3rotab) [reply standalone/pruntime/enclave/src/lib.rs:977]: No. The enclave world and the outside world are different, and they has to be initialized twice

Comment (h4x3rotab) [reply standalone/pruntime/enclave/src/lib.rs:977]: I think the only shared part is the stdout/err channel

**Comment by KiChjang** [comment]
@h4x3rotab Verified that it works locally. Initially I thought it wasn't working, but it was all a matter of setting `RUST_LOG` to `info` to see the output. Pushing a new commit soon.

**Comment by h4x3rotab** [comment]
> @h4x3rotab Verified that it works locally. Initially I thought it wasn't working, but it was all a matter of setting `RUST_LOG` to `info` to see the output. Pushing a new commit soon.
> 
> 
> 
> EDIT: I wonder if that means we should set the default log level at info?

Cool! That would be wonderful

**Comment by h4x3rotab** [comment]
LGTM

**Pull request merged**